scripts/Diffraction/isis_powder/polaris_routines/polaris_algs.py

```diff
@@ -117,6 +123,62 @@ def _obtain_focused_run(run_number, focus_file_path):
     return focused_ws
 
 
+def _generate_grouped_ts_pdf(focused_ws, q_lims):
+    focused_ws = mantid.ConvertUnits(InputWorkspace=focused_ws, Target="MomentumTransfer", EMode='Elastic')
+    min_x = np.inf
+    max_x = -np.inf
+    num_x = -np.inf
+    for ws in focused_ws:
+        x_data = ws.dataX(0)
+        min_x = min(np.min(x_data), min_x)
+        max_x = max(np.max(x_data), max_x)
+        num_x = max(x_data.size, num_x)
+    binning = [min_x, (max_x-min_x)/num_x, max_x]
+    focused_ws = mantid.Rebin(InputWorkspace=focused_ws, Params=binning)
+    focused_data_combined = mantid.ConjoinSpectra(InputWorkspaces=focused_ws)
+    mantid.MatchSpectra(InputWorkspace=focused_data_combined,
+                        OutputWorkspace=focused_data_combined,
+                        ReferenceSpectrum=5)
```

Comment:
Did we pick `5` with some knowledge or a bit of trial and error. This is something we should probably discuss with Helen.

Reply:
spectrum 5 convers the largest range of Q. More information on what the best value for this would be good.

Reply:
Okay. Not for this PR but would it be possible to just determine that from the data itself?

Reply:
certainly, we just need get the range before rebinning

Reply:
Okay. Let's leave that for some future work.